ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/FilterParameterTest.java

```diff
@@ -179,6 +180,68 @@ public void jsonWithBlockAndFromAndToParametersIsInvalid() throws Exception {
     assertThat(parsedFilterParameter.isValid()).isFalse();
   }
 
+  @Test
+  public void jsonWithBlockHashAndFromAndToParametersIsInvalid() throws Exception {
+    final String json =
+        "{\"jsonrpc\":\"2.0\",\"method\":\"eth_getLogs\",\"params\":[{"
+            + "\"address\":\"0x0\", \"fromBlock\": \"0x0\", \"toBlock\": \"pending\","
+            + "\"topics\":["
+            + TOPICS_TWO_THREE_ARRAY
+            + ","
+            + TOPICS_TWO_THREE_ARRAY
+            + "], \"blockHash\": \""
+            + Hash.ZERO
+            + "\"}],\"id\":1}";
+
+    final JsonRpcRequestContext request = new JsonRpcRequestContext(readJsonAsJsonRpcRequest(json));
+    final FilterParameter parsedFilterParameter =
+        request.getRequiredParameter(0, FilterParameter.class);
+
+    assertThat(parsedFilterParameter.isValid()).isFalse();
+  }
+
+  @Test
+  public void jsonWithBlockHashIsValid() throws Exception {
+    final String json =
+        "{\"jsonrpc\":\"2.0\",\"method\":\"eth_getLogs\",\"params\":[{"
+            + "\"address\":\"0x0\","
+            + "\"topics\":["
+            + TOPICS_TWO_THREE_ARRAY
+            + ","
+            + TOPICS_TWO_THREE_ARRAY
+            + "], \"blockHash\": \""
+            + Hash.ZERO
+            + "\"}],\"id\":1}";
+
+    final JsonRpcRequestContext request = new JsonRpcRequestContext(readJsonAsJsonRpcRequest(json));
+    final FilterParameter parsedFilterParameter =
+        request.getRequiredParameter(0, FilterParameter.class);
+
+    assertThat(parsedFilterParameter.isValid()).isTrue();
+    assertThat(parsedFilterParameter.getBlockHash()).isEqualTo(Optional.of(Hash.ZERO));
+  }
+
+  @Test
```

Comment:
The different casing of the tests is a bit confusing again... 

Would it make sense to merge both of these test cases into a `blockHashAliasSucceeds` that asserts the output of both requests is valid and that the output is equal?

Reply:
good idea